search/mongo_search.go

```diff
@@ -429,22 +436,48 @@ func prependCollectionNameToSearchPaths(collectionName string, searchParams []Se
 			// Need to prepend to the OrParam's SearchParam items instead
 			for _, item := range param.Items {
 				searchInfo := item.getInfo().clone()
+
+				if numReferencePaths > 1 {
+					// If we have multiple reference paths we need to duplicate the SearchParamPaths
+					// for each matchable SearchParam so we can test against each $lookup in one $or
+					// clause.
+					duplicatePaths(&searchInfo, numReferencePaths)
+				}
+
 				for i, searchPath := range searchInfo.Paths {
-					searchInfo.Paths[i].Path = prependStr + searchPath.Path
+					searchInfo.Paths[i].Path = prependStr + strconv.Itoa(i) + "." + searchPath.Path
```

Comment:
I don't _think_ this is right...  If `numReferencePaths` is `2` and the original (non-duplicated) `searchInfo` also had 2 paths, then the new `searchInfo` now has 4 paths.  So far so good.  But the logic above will prefix the paths with `_lookup0`, `_lookup1`, `_lookup2`, and `_lookup3`.  But since the `numReferencePaths` is only `2`, there will only be `as` fields for `_lookup0` and `_lookup1`.  Based on how you duplicate the fields (below),  I think you really want to prepend `_lookup0`, `_lookup0` (again for the second `searchInfo` path), `_lookup1`, and `_lookup1` (again, for the second `searchInfo` path).